lib/include/event-driven/vDrawSkin.h

```diff
@@ -0,0 +1,1159 @@
+                }
+            }
+
+        }
+        // return true;
+    }
+
+
+
+
+    std::tuple<double,double> makeMap(int index)const{
+
+        // ID=(ID-3)/12;
+        int ID = getID(index)[0];
+        int taxelIntriangle =getID(index)[1];
+
+        std::vector<double> info = data.find(ID)->second;
+        const double scale=1/40.0;
+
+        double cx = info[0];
+        double cy = info[1];
+        double th = info[2];
+        double gain = info[3];
+        int lrMirror = int (info[4]);
+
+
+        const double CST=cos(DEG2RAD*(th+0));
+        const double SNT=sin(DEG2RAD*(th+0));
+
+        double x=scale*dX[taxelIntriangle];
+        double y=scale*dY[taxelIntriangle];
+        if (lrMirror==1) x=-x;
+
+        double u=cx+CST*x-SNT*y;
+        double v=cy+SNT*x+CST*y;
+        if ((taxelIntriangle!=6) && (taxelIntriangle!=10)) return std :: make_tuple(abs(u), v);
+        else return std :: make_tuple(-1000,-1000);//dummy location
+    };
+
+};
+
+
+
+class Right_hand : public PAD{
+    private:
+    int palmShift;
+//hand must be a wrapper for fingers and palm. Indeed the calibration file is unique
+    public:
+        Right_hand(int shift){
+            xoffset = 352;
+            yoffset = 400;
+            scaling =6;
+            palmShift = shift;
+            noise = 2300;
+            max_value = 10000;
+
+            dX = new double[48];
+            dY = new double[48];
+            std::cout<< "\nInitialise hand"<<std::endl;
+        };
+
+        void initRepresentativeTaxels(std::vector<int> taxelMap){//correct only for 10 pad triangle (?)
+            //yarp::os::RecursiveLockGuard rlg(recursive_mutex);
+            std::list<int> mapp(taxelMap.begin(), taxelMap.end());
+            mapp.sort();
+            mapp.unique();
+
+            size_t mappsize = mapp.size();
+            for (size_t i = 0; i < mappsize; i++)
+            {
+                repr2TaxelList[mapp.front()]=vectorofIntEqualto(taxelMap,mapp.front());
+                mapp.pop_front();
+            }
+
+            int s=0,t=-1;
+
+            for (auto it =repr2TaxelList.begin();it!=repr2TaxelList.end();it++){
+                for (auto it2 =  it->second.begin(); it2 !=it->second.end(); it2++){
+                    if(it->first ==-1)*it2=10000;//dummy value ;
+
+                    if(it->first>60){
+                        //if(s==5) s+=1;/ skip 6th
+
+                        if (t==3){
+                            if(s==7)s+=1;
+                            }
+                        else{
+                            if(s==11) s+=1;//skip 12th except the 4th triangle where it skips the 8th
+                        }
+                        if(s%12==0) {t+=1;s=0;}
+
+                        *it2 =128+s+ t*16;
+                        s+=1;
+
+                    }
+                    else{
+                        int triangleID = (it->first-3)/12;
+                        *it2 += triangleID *4;
```

Comment:
this piece of code is very specific to the current configuration and will require additional work and headaches whenever we will change an epsilon in the taxel configuration

Reply:
agreed - deal with it when someone develops loading from config file